rpcserver.go

```diff
@@ -3299,11 +3299,37 @@ func createRPCOpenChannel(r *rpcServer, graph *channeldb.ChannelGraph,
 		channel.PushAmountSat = uint64(localBalance.ToSatoshis())
 	}
 
-	outpoint := dbChannel.FundingOutpoint
+	if len(dbChannel.LocalShutdownScript) > 0 {
+		_, addresses, _, err := txscript.ExtractPkScriptAddrs(
+			dbChannel.LocalShutdownScript, activeNetParams.Params,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		// We only expect one upfront shutdown address for a channel. If
+		// LocalShutdownScript is non-zero, there should be one payout
+		// address set.
+		if len(addresses) != 1 {
+			return nil, fmt.Errorf("expected one upfront shutdown "+
+				"address, got: %v", len(addresses))
+		}
+
+		channel.CloseAddress = addresses[0].String()
+	}
+
+	// If the server hasn't fully started yet, it's possible that the
+	// channel event store hasn't either, so it won't be able to consume any
+	// requests until then. To prevent blocking, we'll just omit the uptime
+	// related fields for now.
+	if !r.server.Started() {
```

Comment:
imo we should fail the whole request before startup. it's not obvious that some fields will be sporadically omitted

Reply:
Looking further down, it seemed like it was possible for the uptime to be omitted anyway if the `ChannelEventStore` didn't have information about said channel. This should probably never be the case in this scenario, though I'm not sure why the check was added in the first place.

Reply:
I think it's ok to not have these fields if the server hasn't started, because it'll have a zero `lifetime` value which tells the user that we have not monitored uptime for the peer yet? 

We check this error in case there's a race between the event store being notified of a new channel and a query to `ListChannels`, pretty unlikely in regular use, but the itests hit it when opening channels and checking for their presence in the list.

Reply:
> We check this error in case there's a race between the event store being notified of a new channel and a query to ListChannels

What exactly is the race? If we got the notification, then shouldn't 1) the database already have been modified, and 2) the server already be started?

That flow seems perfectly reasonable, if we hit it in our itests i expect people will hit it in production. Is the issue then that we send out the event before the database has finished updating?

Reply:
Right now the order of events is:
1. We mark a channel as fully open when the open tx confirms
2. Funding manager notifies that the channel is open
3.  Channel notifier receives notification and adds channel to store + starts tracking uptime

If `ListChannels` is called between (2) and (3), the uptime query will fail because the store doesn't know that the channel exists yet. The itests hit this because we open a channel, mine then call list channels to make sure it's there. But as you say this is a pretty normal flow, so we [silence](https://github.com/lightningnetwork/lnd/blob/444eec2abc1dbf550c5edde6d2845fc74de1c56d/rpcserver.go#L3310) the error; otherwise `ListChannels` would fail. When this was written, it was deemed acceptable to omit the fields because you can figure out that the channel has not yet been monitored using the lifespan field. 

All of the above is unrelated to the server being started (it is started in the above scenario), it's just an answer to @wpaulino 's comment:
> it seemed like it was possible for the uptime to be omitted anyway if the ChannelEventStore didn't have information about said channel .... though I'm not sure why the check was added in the first place.

And re this comment
> What exactly is the race? If we got the notification, then shouldn't 1) the database already have been modified, and 2) the server already be started?

This race can only be present if the server is started. It's unrelated to the current issue of the server not being up yet which blocks calls. The solution for the server not being started yet is to omit these fields when unavailable, which was already the case in some situations.

I hope that clears this up, my original change should have been more clearly commented.

Reply:
Awesome, okay it is clear to me now. Thanks for the detailed response!